### DISCUSSION TITLE
How to customise template mutator and acessor?

### DISCUSSION THREAD

**Discussion by brunojoao** [phpactor/phpactor#2711] [Q&A] [answered]
I try to use mutator and accessor

![image](https://github.com/user-attachments/assets/53f567e5-c7ac-4109-84d2-f25f6a81bc33)

but it is generating it wrong, without the r in the integer, I also want it to work in the php5.6 style, without, I already tried in ~/config/phpactor/templates/mutator. I've already tried .twig, .php.twig, nothing works, I'm having to generate and replace.

[phpactor.json](https://github.com/user-attachments/files/16445986/phpactor.json)
config file

**Comment by dantleech** [comment]
It should be 'int' not integer 🤔 I assume you have "@var integer" off-screen?

**Reply by brunojoao** [reply]
I'm on an old project, I needed to use PHP 5.6 style, without typing, would it be possible? customizing template or something like that, I tried different ways to call the template but it didn't work, while in r, everything is ok, it could have been anything else

**Reply by dantleech** [reply, marked as answer]
yes I think you can:

https://github.com/phpactor/phpactor/blob/master/templates/code/Method.php.twig

copying that and customising it in `.phpactor/templates/Method.php.twig` should do it.

We should probably add a command to automatically copy all the templates into the folder so they can be easily and reliably customised.

**Comment by brunojoao** [comment]
**This is the code in case anyone needs it for an older version of php:**
~/.config/phpactor/templates/Method.php.twig

```

{# Method.php.twig #}
{% if prototype.docblock.notNone %}
/**
{% for line in prototype.docblock.asLines %}
 * {{ line|raw }}
{% endfor %}
 */
{% endif %}
{% if prototype.isAbstract %}abstract {%endif %}{{ prototype.visibility }} {% if prototype.isStatic %}static {%endif %}function {{ prototype.name}}({% spaceless %}
{% for parameter in prototype.parameters %}{{ generator.render(parameter, variant)|raw }}{% if not loop.last %}, {% endif %}{% endfor %}{% endspaceless %})
{% if prototype.isStatic %}
{% if prototype.docblock.notNone %}
/**
 * @var {{ prototype.returnType.type|default('mixed') }}
 */
{% endif %}
{% endif %}


```

However, mutators still come with typing, it would be so good if there was an easier way xD
